certbot/tests/cli_test.py

```diff
@@ -63,7 +65,8 @@ def _help_output(self, args):
         return output.getvalue()
 
     @mock.patch("certbot.cli.flag_default")
-    def test_cli_ini_domains(self, mock_flag_default):
+    @test_util.patch_get_utility()
```

Comment:
If that still causes `_stdout_notification` to write to the correct stdout (e.g. the `StringIO` object and not the real stdout), something like that is totally fine.

I accidentally left out the crucial detail of my plan earlier to have `_help_output` call `_unmocked_parse`. The thought was `_help_output` is the only one that actually has to set the `side_effect` of the `notification` method as it's specifically testing the program output. After steps 1 and 2, step 3 would look something like:
```
     def _help_output(self, args):
         "Run a command, and return the output string for scrutiny"
-
         output = six.StringIO()
+
+        def write_msg(message, *args, **kwargs):
+            output.write(message)
+
         with mock.patch('certbot.main.sys.stdout', new=output):
-            with mock.patch('certbot.main.sys.stderr'):
-                self.assertRaises(SystemExit, self.parse, args, output)
+            with test_util.patch_get_utility() as mock_get_utility:
+                mock_get_utility().notification.side_effect = write_msg
+                with mock.patch('certbot.main.sys.stderr'):
+                    self.assertRaises(SystemExit, self._unmocked_parse, args, output)
         return output.getvalue()
```
I have a slight preference for this approach, but if what you have above works, that's fine with me. Sorry for the misleading post!